x-pack/plugins/observability/public/components/shared/exploratory_view/configurations/infra_logs/kpi_over_time_config.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ConfigProps, SeriesConfig } from '../../types';
+import { FieldLabels, RECORDS_FIELD, ReportTypes } from '../constants';
+import { LOG_RATE as LOG_RATE_FIELD } from '../constants/field_names/infra_logs';
+import { LOG_RATE as LOG_RATE_LABEL } from '../constants/labels';
+
+export function getLogsKPIConfig(configProps: ConfigProps): SeriesConfig {
+  return {
+    reportType: ReportTypes.KPI,
+    defaultSeriesType: 'bar',
+    seriesTypes: [],
+    xAxisColumn: {
+      label: i18n.translate('xpack.observability.exploratoryView.logs.logRateXAxisLabel', {
+        defaultMessage: 'Timestamp',
+      }),
+      dataType: 'date',
+      operationType: 'date_histogram',
+      sourceField: '@timestamp',
+      isBucketed: true,
+      scale: 'interval',
+    },
+    yAxisColumns: [
+      {
+        label: i18n.translate('xpack.observability.exploratoryView.logs.logRateYAxisLabel', {
```

Comment:
Hmm, I’m not sure. I’d need to check if those custom series names are passed through as part of the Lens attributes.

Reply:
They should be passed AFAIK, it's tricky in case of breakdown though.
Worth raising an issue for this.

Reply:
@dominiqueclarke @liciavale Do you remember this, did we revert this change? or is it broken?

Legend should reflect series name, no?

Reply:
Hmm, so I thought this change might work:

```ts
const { breakdown, name } = layerConfig;
const { sourceField, operationType, label } = layerConfig.seriesConfig.yAxisColumns[0];

if (sourceField === RECORDS_FIELD || !sourceField) {
  return this.getRecordsColumn(name ?? label);
}
```

it takes `name` off the `layerConfig` and passes it through when dealing with a records field. However it leads to a problem as there's always a default `name` (which is ugly in comparison to the normal labels, in this case "Log rate per minute"). I guess you could do some check to see if the `dataType` exists in the `name` but there's nothing stopping someone setting a custom name that contains it either. I think this requires a proper followup issue.

Here's a visual example:

![Screenshot 2022-03-18 at 18 13 37](https://user-images.githubusercontent.com/471693/159060544-6925e2d9-b1d3-46c7-9f5d-4f4edbcc4dba.png)

Reply:
Maybe the default name could be prettier in the first place?

Reply:
Possibly, but it has cascading effects as `infra_logs` isn't just an Exploratory View naming but something from the overall Observability plugin codebase. So it would have to be changed all the way up the hierarchy. I guess there could be another map, `dataTypeToSeriesName` or something similar, but I'm not sure that's a great idea either.